examples/vision/3D_image_classification.py

```diff
@@ -0,0 +1,454 @@
+    y_train_aug = []
+
+    # rotate all scans
+    for volume, label in zip(inputs, labels):
+        volume, label = augment_data(volume, label, rotate)
+        volume = normalize(volume)
+        x_train_aug.append(volume)
+        y_train_aug.append(label)
+
+    # gaussian Blur all scans
+    for volume, label in zip(inputs, labels):
+        volume, label = augment_data(volume, label, blur)
+        volume = normalize(volume)
+        x_train_aug.append(volume)
+        y_train_aug.append(label)
+
+    x_train_aug = np.array(x_train_aug)
+    y_train_aug = np.array(y_train_aug)
+    return x_train_aug, y_train_aug
+
+
+"""
+Augmented the training data
+"""
+
+x_train_aug, y_train_aug = get_augmented_data(x_train, y_train)
+print(x_train_aug.shape, y_train_aug.shape)
+
+"""
+Normalize the training and test data subsets
+"""
+
+# Normalize data
+x_train = np.array([normalize(x) for x in x_train])
+x_test = np.array([normalize(x) for x in x_test])
+
+"""
+Visualize an augmented CT scan
+"""
+
+import matplotlib.pyplot as plt
+
+img = x_train_aug[20]
+print("Dimension of the CT scan is:", img.shape)
+print(np.max(img), np.min(img))
+plt.imshow(img[:, :, 80], cmap="gray")
+
+"""
+Merge the augmented and training data. Note that the augmented data is not normalize
+here, as it was already normalized right after the augmentation.
+"""
+
+# Merge raw data and augmented data
+x_all = np.concatenate((x_train_aug, x_train), axis=0)
+x_all = np.expand_dims(x_all, axis=4)
+y_all = np.concatenate((y_train_aug, y_train), axis=0)
+print("Shape of upsampled training data inputs and labels: ", x_all.shape, y_all.shape)
+
+"""
+## Define 3D convolutional neural network
+
+To make the model easier to understand, blocks are defined. Since this is a
+3D CNN, 3D convolutions are used. The architecture of the 3D CNN used in this example
+is based on this [paper](https://arxiv.org/abs/2007.13224).
+"""
+
+
+def get_model(img_size, num_classes):
+    # build a 3D convolutional neural network model
+    inputs = keras.Input(img_size)
+
+    x = layers.Conv3D(filters=16, kernel_size=3, activation="relu")(inputs)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=32, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=64, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=128, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Flatten()(x)
+    x = layers.Dense(units=256, activation="relu")(x)
+    x = layers.Dropout(0.1)(x)
+
+    outputs = layers.Dense(units=2, activation="softmax")(x)
+
+    # define the model
+    model = keras.Model(inputs, outputs)
+    return model
+
+
+# free up RAM
+keras.backend.clear_session()
```

Comment:
You don't need this in this example, it will clear very little. It's only justified if you have a for loop where you create dozens of models

Reply:
Removed!